sphinx_gallery/tests/test_gen_gallery.py

```diff
@@ -339,6 +345,45 @@ def test_example_sorting_title(sphinx_app_wrapper):
     _check_order(sphinx_app, "title")
 
 
+@pytest.mark.add_conf(
+    content="""
+from sphinx_gallery.sorting import FunctionSortKey
+
+ORDER = [
+    "plot_3.py",
+    "plot_2.py",
+    "plot_1.py",
+]
+
+def custom_sorter(filename):
+    return ORDER.index(filename)
+
+sphinx_gallery_conf = {
+    "examples_dirs": "src",
+    "gallery_dirs": "ex",
+    "within_subsection_order": FunctionSortKey(custom_sorter),
```

Comment:
I can add here 
```py
    "minigallery_sort_order": FunctionSortKey(custom_sorter),
    "subsection_order": FunctionSortKey(custom_sorter),
```
and the test will still pass, but I think that's not especially informative because (IIUC) the site being built by for this test doesn't have subsections / minigalleries? Could use some guidance on how/where to test that `FunctionSortKey(my_func)` and FQN strings work correctly for minigalleries and subsection ordering... maybe that's already tested here?:

https://github.com/sphinx-gallery/sphinx-gallery/blob/9947c26a7457f7ef35b41cee7d6fa89e1f36c67d/sphinx_gallery/tests/test_full.py#L1350

...but I find that test/parametrization inscrutable, so I'd appreciate some guidance on whether those cases are already covered, and if not, how/where I should add/tweak a test to cover them.

Reply:
So we could do minigalleries here.

I added a `add_rst` pytest mark here, which will add a `index.rst` file in src with the contents of `file` arg. (This happens in the `sphinx_wrapper_app` in conftest). See this test for example:

https://github.com/sphinx-gallery/sphinx-gallery/blob/13f7e06c7e70cd4c3fa04ab7c317c4faf4f2ce26/sphinx_gallery/tests/test_gen_gallery.py#L647-L662

AFAICT we only test FQN string for `subsection_order` in tinybuild.
It may be a big ask, but what do you think about adding a second subsection to the `sphinx_gallery/tests/testconfs/src` gallery...?